website/src/parsers/js/babel-eslint-parser.js

```diff
@@ -13,13 +13,16 @@ export default {
   locationProps: new Set(['loc', 'start', 'end', 'range']),
 
   loadParser(callback) {
-    require(['@babel/eslint-parser'], callback);
+    require(['@babel/eslint-parser', '@babel/plugin-syntax-jsx'], callback);
   },
 
   parse(parser, code) {
     const opts = {
       sourceType: 'module',
       requireConfigFile: false,
+      babelOptions: {
+        presets: ['@babel/plugin-syntax-jsx'],
```

Comment:
> Is it necessary to require the plugin in loadParser? I required it just in case it was necessary, but maybe Babel is loading it by itself anyway just with the string from the config?

Have you tested it? (that's not a trick question 😆). I don't know what exactly babel does to resolve presets by strings (I guess we could find out), but I wouldn't expect this to work (usually those strings would be resolved to something on disk). But maybe babel is doing it in a way that's compatible with webpack... still I would probably pass the reference to the plugin to the options instead of referring to it by name.

However, having set all that and looking at how we configure things for the other babel parsers, it made me realize that all of this is unnecessary. It should work by directly enabling the JSX plugin in the parser without requiring any additional modules (because JSX support is built-into the parser; in fact, all that the syntax plugin does is [setting that option](https://github.com/babel/babel/blob/b91c3e15738ca9a390135bdce9d845da90c40733/packages/babel-plugin-syntax-jsx/src/index.ts#L20)):

```js
const opts = {
  // ...
  babelOptions: {
    parserOpts: {
      plugins: ['jsx'],
    }
  },
};
```

See https://babeljs.io/docs/en/options#parseropts and https://babeljs.io/docs/en/babel-parser#language-extensions

Reply:
> still I would probably pass the reference to the plugin to the options instead of referring to it by name

Yeah I wanted to do this at the beginning, but after doing a bit of research, I couldn't see any examples where I could pass a `require`d plugin to the config.

Reply:
> Have you tested it?

Nope, haven't even cloned the repo - just doing this all by manually editing files on GitHub.

Reply:
Good point about the `parserOpts` though, that could be also a way to go. I'll make the change.